cabal-install/Distribution/Client/ProjectPlanning.hs

```diff
@@ -731,6 +733,11 @@ getPkgConfigDb verbosity progdb = do
     liftIO $ readPkgConfigDb verbosity progdb
 
 
+getDirectoryContentsMonitored :: FilePath -> Rebuild [FilePath]
+getDirectoryContentsMonitored dir = do
+    monitorFiles [monitorDirectory dir]
```

Comment:
From the `stat(2)` man page (and I think this is true on Windows too):

> Moreover, st_mtime of a directory is changed by the creation or deletion of files in that directory.

So this should be just enough for `getDirectoryContentsMonitored`, since it's the creation of dirs in the store that should trigger re-reading the entries of that dir.

And `getDirectoryContentsMonitored` should be just enough for reading the store, since entries in the store are (supposed to be) immutable so it's just the adding/removing of them that matters.

So where is this breaking down?